projects/graphql-client/src/graphql-apollo-integration.test.ts

```diff
@@ -169,28 +178,7 @@ describe('GraphQl Apollo Integration Service', () => {
     flush();
   }));
 
-  // test('does not cache entities, even if they share an id', fakeAsync(() => {
```

Comment:
what happened to this test? were we unable to make it work?

Reply:
This is not a valid test. Why would we want to disable caching for this fetch by id? The default behavior would merge the two response if we ask for new fields. Then only time it will refetch is when the request param like time, filters etc changes.

Reply:
That seems correct behavior to me. So why do we not want to cache them?

Reply:
if they're requested at different times (hence the queryImmediately behavior in existing test) they wouldn't merge - so it's testing what you're describing, that it refetches on a param change.

Reply:
Yes. But in the test we also fire the old request again and expect to get the previously cached data for the same id. Since Apollo got newer data for the id, old info will be overwritten. Only this part of the test doesn't seem valid to me. 

```
  modifiedResponse.data.test.value = 'baz';
     spectator.expectOne(graphQlUri, HttpMethod.POST).flush(modifiedResponse);
     // Now make the first request again - it should be the original result
     spectator.service.queryImmediately(buildRequest()).subscribe(result => {
      expect((result as { value: string }).value).toEqual('bar');
    });
    // Tick, our responses are always async//   modifiedResponse.data.test.value = 'baz';
     spectator.expectOne(graphQlUri, HttpMethod.POST).flush(modifiedResponse);
    // Now make the first request again - it should be the original result
    spectator.service.queryImmediately(buildRequest()).subscribe(result => {
      expect((result as { value: string }).value).toEqual('bar');
     });
    // Tick, our responses are always async
```

Reply:
Okay hmm.. I misunderstood the test.

Reply:
Yes. We need to change the response for this test though.  Apollo looks at result set and caches it according to the object type and defined type policies. For an array response, default behavior is to keep request-response pair as unique. So any change in request will trigger new http call. It achieves this by setting `[]` as the id for array response. Then it goes inside the array, and caches each individual object according to defined type policies or by simply looking for presence of `id`(default). 

When we make a fetchById query, we will only pass its id in the argument and nothing else. E.g `actor(id){...fields}`.
Apollo would use this id to look for cached data, compare and merge and use it for future request.

In the current test, we are returning an object with id for a query that has not id in argument. 

Fixed this by returning an array to test first scenario. Added another test for second scenario.

Reply:
Done